httpx/dispatch/http2.py

```diff
@@ -108,18 +110,19 @@ def initiate_connection(self) -> None:
     async def send_data(
         self, stream_id: int, data: bytes, timeout: TimeoutConfig = None
     ) -> None:
-        flow_control = self.h2_state.local_flow_control_window(stream_id)
-        chunk_size = min(len(data), flow_control)
-        for idx in range(0, len(data), chunk_size):
-            chunk = data[idx : idx + chunk_size]
-
-            logger.debug(
-                f"send_data stream_id={stream_id} data=Data(<{len(chunk)} bytes>)"
+        while data:
+            flow_control = self.h2_state.local_flow_control_window(stream_id)
+            chunk_size = min(
+                len(data), flow_control, self.h2_state.max_outbound_frame_size - 1
```

Comment:
I'm not sure that here we need to reduce the value of `max_outbound_frame_size` by one, because according to the `h2` documentation, this is the [maximum value](https://python-hyper.org/projects/h2/en/stable/api.html#h2.connection.H2Connection.max_outbound_frame_size) that can be sent and it [should not](https://python-hyper.org/projects/h2/en/stable/api.html#h2.connection.H2Connection.send_data) cause an error in the form in which it is given to us from `h2` state machine.

Reply:
In testing an error was thrown when chunk size was `self.h2_state.max_outbound_frame_size` (which appeared to always be 2^14 = 16384) which said that the limit on frame size was the 16383. Adding in the -1 fixed the issue, and I assumed that somewhere the maximum value was being interpreted as strictly greater than as opposed to greater than or equal to.

Reply:
I think that this error was caused by unhandled `WindowsUpdated` event and error in splitting data into chunks. In `h2` this [checked](https://github.com/python-hyper/hyper-h2/blob/385ced7d30a7ccb9b3f163f1bc4e90cf2f4ee184/h2/connection.py#L850) as larger then, so I don't think that there is a need for reducing value.

Reply:
Removed -1 in new commit.